lms/djangoapps/branding/api.py

```diff
@@ -0,0 +1,289 @@
+            ("faq", marketing_link("FAQ"), _("FAQs")),
+            ("contact", marketing_link("CONTACT"), _("Contact")),
+            ("jobs", marketing_link("JOBS"), _("Jobs")),
+            ("donate", marketing_link("DONATE"), _("Donate")),
+            ("sitemap", marketing_link("SITE_MAP"), _("Sitemap")),
+        ]
+        if link_url and link_url != "#"
+    ]
+
+
+def _footer_legal_links():
+    """Return the legal footer links (e.g. terms of service). """
+    return [
+        {
+            "name": link_name,
+            "title": link_title,
+            "url": link_url,
+        }
+        for link_name, link_url, link_title in [
+            ("terms_of_service", marketing_link("TOS"), _("Terms of Service")),
+            ("honor_code", marketing_link("HONOR"), _("Honor Code")),
+            ("terms_of_service_and_honor_code", marketing_link("TOS_AND_HONOR"), _("Terms of Service & Honor Code")),
+            ("privacy_policy", marketing_link("PRIVACY"), _("Privacy Policy")),
+            ("accessibility_policy", marketing_link("ACCESSIBILITY"), _("Accessibility Policy")),
+        ]
+        if link_url and link_url != "#"
+    ]
+
+
+def _footer_mobile_links(is_secure):
+    """Return the mobile app store links.
+
+    Args:
+        is_secure (bool): Whether the request is using TLS.
+
+    Returns: list
+
+    """
+    mobile_links = []
+    if settings.FEATURES.get('ENABLE_FOOTER_MOBILE_APP_LINKS'):
+        mobile_links = [
+            {
+                "name": "apple",
+                "title": "Apple",
+                "url": settings.MOBILE_STORE_URLS.get('apple', '#'),
+                "image": _absolute_url_staticfile(is_secure, 'images/app/app_store_badge_135x40.svg')
+            },
+            {
+                "name": "google",
+                "title": "Google",
+                "url": settings.MOBILE_STORE_URLS.get('google', '#'),
+                "image": _absolute_url_staticfile(is_secure, 'images/app/google_play_badge_45.png')
+            }
+        ]
+    return mobile_links
+
+
+def _footer_logo_img(is_secure):
+    """Return the logo used for footer about link
+
+    Args:
+        is_secure (bool): Whether the request is using TLS.
+
+    Returns:
+        Absolute url to logo
+    """
+    logo_name = (
+        u"images/edx-theme/edx-header-logo.png"
+        if settings.FEATURES.get('IS_EDX_DOMAIN', False)
+        else settings.FOOTER_ORGANIZATION_IMAGE
+    )
+
+    return _absolute_url_staticfile(is_secure, logo_name)
+
+
+def _absolute_url(is_secure, url_path):
+    """Construct an absolute URL back to the site.
+
+    Arguments:
+        is_secure (bool): If true, use HTTPS as the protocol.
+        url_path (unicode): The path of the URL.
+
+    Returns:
+        unicode
+
+    """
+    protocol = "https://" if is_secure else "http://"
+    site_name = microsite.get_value('SITE_NAME', settings.SITE_NAME)
+    return u"{protocol}{site_name}{url_path}".format(
```

Comment:
i think we should use the [urlparse module](https://docs.python.org/2/library/urlparse.html) for manipulations like this, unless there's a good reason not to.